### DISCUSSION TITLE
EditableLargeNumberMenuItem: Change color of the edit cursor

### DISCUSSION THREAD

**Discussion by futurexdesign** [davetcc/tcMenu#338] [General]
When editing an `EditableLargeNumberMenuItem`,  there is a cursor under the current digit being edited.   This cursor appears as a white line.   The UI I am working with uses selection colors of white background/ black text.   As such, the cursor is invisible on the screen, as it is also rendered in white. 

I've tried to figure out how to change the color of this cursor, but haven't yet found a way to do so.   It doesn't appear to be affected by any of the 4 colors in the palette assigned to the menu item. it always seems to render in white. 

Is there a way to change this color?

**Comment by davetcc** [comment]
The cursor at the moment is fixed as an underline, but there is a plan to support a box cursor as well. However, the colours can be set in your theme or even main class you will need to use the namespace tcgfx:

    // near the top somewhere
    using namespace tcgfx;

    // then either in the theme file, or somewhere else during setup (if you are in the theme file you won't need to get off the renderer
    // the two color parameters are cursor color and text color.
    renderer.getGraphicsPropertiesFactory().setEditCursorMode(CURSOR_MODE_UNDERLINE, RGB(0,0,0), RGB(1,1,1))

For more info on how it is drawn, the starting point is:

https://github.com/davetcc/tcMenuLib/blob/master/src/graphics/GraphicsDeviceRenderer.cpp#L162
